catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/json/MetacardTypeSerializer.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.source.solr.json;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.MetacardType;
+import java.io.IOException;
+
+public class MetacardTypeSerializer extends StdSerializer<MetacardType> {
+
+  public MetacardTypeSerializer() {
+    super(MetacardType.class);
+  }
+
+  @Override
+  public void serialize(MetacardType value, JsonGenerator gen, SerializerProvider provider)
+      throws IOException {
+    gen.writeStartObject();
+    gen.writeStringField("name", value.getName());
+    gen.writeArrayFieldStart("attributeDescriptors");
+    for (AttributeDescriptor attributeDescriptor : value.getAttributeDescriptors()) {
+      gen.writeStartObject();
+      gen.writeStringField("name", attributeDescriptor.getName());
+      gen.writeBooleanField("indexed", attributeDescriptor.isIndexed());
+      gen.writeBooleanField("stored", attributeDescriptor.isStored());
+      gen.writeBooleanField("tokenized", attributeDescriptor.isTokenized());
+      gen.writeBooleanField("multiValued", attributeDescriptor.isMultiValued());
+      gen.writeStringField(
+          "attributeFormat", attributeDescriptor.getType().getAttributeFormat().toString());
```

Comment:
✏️ Such a big assumption on the map creation in `BasicTypes` that the format's string + "_TYPE" is equal to the key used when that key is actually hardcoded ...

Reply:
I thought the same thing.  Looks like that function was added with the validation json files and that string is a value used in those files.  Sadly that accidental coupling of implementation details got through review.  I will see if I can find an alternative for this PR.

Reply:
I went ahead and fixed `BasicTypes` to not be coupled to `ValidationParser` implementation details.